### DISCUSSION TITLE
Accessing a local database from script scheduled in airflow(on docker)

### DISCUSSION THREAD

**Discussion by dominikhei** [apache/airflow#26651] [Q&A] [answered]
Hello, 

I want to schedule a local script via airflow(on docker) which connects to a local mysql database via mysql.connector. 
However it never works. The general error is always can't connect to the database. 

The python script is locally and not containerized. The database port is 3306 and localhost. When I specify localhost in the mysql connector fields, the dag can not even be Imported. If I specify host.docker.internal I can import the dag but still can not connect to the database. 

`db = mysql.connector.connect(
        user = 'root',
        password = '________',
        port='3306'
        host='localhost',
        database='_______'
        )`

Has someone got a solution for this problem? I don't know how to access the local database when the script is scheduled by airflow.

**Comment by Taragolis** [comment, marked as answer]
This is general question "How to connect from a container to a service on the host" not related to Airflow itself and relate to docker.

Anyway. You can not use `localhost` DNS record inside the docker to connect to host, because inside a docker container it usual reference to container itself.

This links might be helpful:
- https://docs.docker.com/desktop/networking/#i-want-to-connect-from-a-container-to-a-service-on-the-host
- https://stackoverflow.com/a/43541732
